src/api/procedures/investInSto.ts

```diff
@@ -31,10 +31,53 @@ export type Params = InvestInStoParams & {
  * @hidden
  */
 export interface Storage {
-  investmentPortfolioId: PortfolioId;
+  purchasePortfolioId: PortfolioId;
   fundingPortfolioId: PortfolioId;
 }
 
+/**
+ * @hidden
+ */
+type TierStats = {
+  remaining: BigNumber;
+  price: BigNumber;
+  remainingAmount: BigNumber;
+};
+
+/**
+ * @hidden
+ */
+export const calculateTierStats = (
+  tiers: Tier[],
+  purchaseAmount: BigNumber,
+  maxPrice?: BigNumber
+): TierStats => {
+  return tiers.reduce<TierStats>(
+    (prev, { remaining, price }) => {
```

Comment:
Destructure prev so you don't have to do `prev.remainingAmount` multiple times (same goes for other properties)